lib/adapters/FileSystem.js

```diff
@@ -26,16 +26,25 @@ class FileSystem extends AbstractAdapter {
 	 * The Constructor.
 	 *
 	 * @param {object} parameters Parameters
-	 * @param {string} parameters.virBasePath Virtual base path
-	 * @param {string} parameters.fsBasePath (Physical) File system path
+	 * @param {string} parameters.virBasePath
+	 *   Virtual base path. Must be absolute, POSIX-style, and must end with a slash
+	 * @param {string} parameters.fsBasePath
+	 *   File System base path. Must be absolute and must use platform-specific path segment separators
 	 * @param {string[]} [parameters.excludes] List of glob patterns to exclude
 	 * @param {object} [parameters.useGitignore=false]
 	 *   Whether to apply any excludes defined in an optional .gitignore in the given <code>fsBasePath</code> directory
 	 * @param {@ui5/project/specifications/Project} [parameters.project] Project this adapter belongs to (if any)
 	 */
 	constructor({virBasePath, project, fsBasePath, excludes, useGitignore=false}) {
 		super({virBasePath, project, excludes});
-		this._fsBasePath = path.resolve(fsBasePath);
+
+		if (!fsBasePath) {
+			throw new Error(`Unable to create adapter: Missing parameter 'fsBasePath'`);
+		}
+
+		// Ensure path is resolved to an absolute path, ending with a slash (or backslash on Windows)
+		// path.resolve will always remove any trailing segment separator
+		this._fsBasePath = path.join(path.resolve(fsBasePath), path.sep);
```

Comment:
here we remove a trailing slash any way... I see

Reply:
Correct. Also, there are already many consumers omitting the trailing slash so I didn't want to add this restriction. We just resolve it ourselves (which is anyways necessary, since the path might be relative too 🙈 ) and everything should be fine and compatible.

Reply:
Ack